### PR TITLE
Bug 1235007 - Add integration test: |lockscreen_camera_button_test.js| to test camera dismissing in secure mode

### PR DESCRIPTION
**Pull request by snowmantw**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=dd4170a79f2849f2ea7f7a50bc39fb051e8251b3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=90a01a8e660609540c467ba3e6f15fb7fc51d6a8

File: apps/system/test/marionette/lockscreen_camera_button_test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    screenLockPanel.toggleScreenLock();
+    screenLockPanel.togglePasscodeLock();
+    screenLockPanel.typePasscode('1234', '1234');
+    screenLockPanel.tapCreatePasscode();
+    settingsApp.close();
+    lockScreen.lock();
+  });
+
+  test('slide to invoke passcode panel, then press the camera button ' +
+       ' to launch the camera app',
```
Comment (timdream) [inline apps/system/test/marionette/lockscreen_camera_button_test.js]: ... the secure camera app

File: apps/system/test/marionette/lockscreen_camera_button_test.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        } else {
+          return false;
+        }
+      });
+    })
+    .then(done)
+    .catch(done);
+  });
+
+  test('after launch the app, press power button twice should dismiss it',
```
Comment (timdream) [inline apps/system/test/marionette/lockscreen_camera_button_test.js]: `press power button twice` -> `turn off and turn on screen`?

File: apps/system/test/marionette/lockscreen_camera_button_test.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        if (theCameraApp) {
+          return theCameraApp.displayed();
+        } else {
+          return false;
+        }
+      });
+    })
+    .then(function() {
+      client.executeScript(function() {
+        window.wrappedJSObject.dispatchEvent(new CustomEvent('home'));
```
Comment (timdream) [inline apps/system/test/marionette/lockscreen_camera_button_test.js]: I am not entire sure if there is better way to similar this (send mocked key event from Gecko?), but we should do that if there is ...

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9e284dea1fe143cc04994320c3710262e5f097cd

File: apps/system/test/marionette/lockscreen_camera_button_test.js (hunk 3 above)
Comment (snowmantw) [inline apps/system/test/marionette/lockscreen_camera_button_test.js]: From MarionetteJs API document I haven't found any high-level API to do that, unless I dig into Marionette.CommandStream to send raw Marionette command. I don't know, for this case I don't feel it's worth to do, but if we want to build an abstraction for hardware like Gip's `driver` then it's a must-have.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4d2c24689b6d83a903cac7bec65278c08a9f46a9

**Pull request merged**